pagopa/io-app
# chore: [IOCOM-1357] Entry point, structure and first half of saga for FIMS implementation

## Short description
first steps for FIMS app implementation
<img width="250" alt="Screenshot 2024-05-14 at 12 28 44" src="https://github.com/pagopa/io-app/assets/60693085/48f84ca3-3857-4628-8d72-1ebf1dc33e1b">


## List of changes proposed in this pull request
- marked old FIMS feature as legacy
- added new FIMS boilerplate (navigator, reducer, saga, selectors)
- added new FIMS testing screen -- the one seen in the image, plus its loading state
- added getConsentsList saga, based on mocks gently provided by @Vangaorth 
- tests to come

also this PR includes part of the second half of the saga, which will be fully developed in the next one.

## How to test
using the `dev-server`, checkout the [feature/fims](https://github.com/pagopa/io-dev-api-server/tree/feature/fims) branch, then in `config.ts` set the `withCTA` parameter to true.
A FIMS message should appear in your messge's list, interact with it and click on its CTA, the portrayed screen should show.
